src/server/main_service.cc

```diff
@@ -2708,9 +2725,15 @@ void Service::Register(CommandRegistry* registry) {
       << CI{"EVAL", CO::NOSCRIPT | CO::VARIADIC_KEYS, -3, 3, 3, acl::kEval}
              .MFUNC(Eval)
              .SetValidator(&EvalValidator)
+      << CI{"EVAL_RO", CO::NOSCRIPT | CO::VARIADIC_KEYS, -3, 3, 3, acl::kEvalRo}
```

Comment:
Should we have `CO::READONLY` for this and `EVALSHA_RO` below?

Reply:
Do we use `CO::READONLY` (I think we do but it's worth checking in transaction files)? If so, we have a `small issue`. Vlad introduced `implicit acl categories` https://github.com/dragonflydb/dragonfly/pull/3411/files#diff-50ecdb571d03f8f9d1d13b16446b808f929af764660b924f0413f554cc653b43R55 so the problem is that it will add `READ` which is not part of `EVAL_RO`. If we need `CO::READONLY` make sure we don't add this implicitly for this case

Reply:
Is there any harm in adding `READ` to these though?

Reply:
We do use `CO::READONLY`, for a few things. We do not allow to run such commands on replica, we don't send them from masters to replica, it's also used in our locking system (shared vs exclusive).

Reply:
Uh then yes we should include this and take care of implicit ACL's

Reply:
For some reason redis dones't mix readonly and scripts

https://github.com/redis/redis/blob/701f06657d20ec42f0cf78de0ac9d7197e44c00c/src/server.c#L3044

Yes, we can exclude it even with implicit acl

Reply:
So just to check I understand, I should set `CO::READONLY`?

Then whats needed from ACLs? `EVAL_RO`/`EVALSHA_RO` already explicitly define the ACL categories so no changes are needed right?